src/dbnode/server/server.go

```diff
@@ -555,12 +555,14 @@ func Run(runOpts RunOptions) {
 		protoEnabled = true
 	}
 	schemaRegistry := namespace.NewSchemaRegistry(protoEnabled, logger)
-	// TODO [haijun] remove after PR to set schema in etcd is done (plan layed out in issue #1614).
-	// To unblock #1578, we will load user schema from db node configuration into schema registry
-	// at dbnode startup/initialization time, there will be no dynamic schema update.
+	// For application m3db client integration test convenience (where a local dbnode is started as a docker container),
+	// we allow loading user schema from db node configuration into schema registry
+	// at dbnode startup/initialization time.
 	if protoEnabled {
 		for nsID, protoConfig := range cfg.Proto.SchemaRegistry {
```

Comment:
This logic means that if you have anything put under the `SchemaRegistry` field we're ignore the schemas in etcd, but if you don't put anything, we'll try and get a schema out of etcd. Is that correct?

What happens if we are `protoEnabled`, don't set the schema registry, but there is no schema in etcd? Will the dbnodes error out somehow? Just want to understand how all this config flows together cause it seems like `protoEnabled` might have two purposes now

Reply:
yes, if proto is enabled, but schema is not populated prior to db/namespace initialization, namespace won't be initialized. We log an error and move on with other namespaces. 

This part of logic is in the previous PR. I believe you had the same comment and I responded there already.

Reply:
Haha ok well at least I'm consistent :p Sorry hard to keep track of everything.